airflow/providers/cncf/kubernetes/utils/pod_manager.py

```diff
@@ -180,7 +180,7 @@ def follow_container_logs(self, pod: V1Pod, container_name: str) -> None:
             So the looping logic is there to let us resume following the logs.
         """
 
-        def follow_logs(since_time: Optional[datetime] = None) -> Optional[datetime]:
+        def follow_logs(since_time: Optional[datetime] = None) -> Union[Date, Time, Duration, datetime, None]:
```

Comment:
That's hell of a union :)

Reply:
The result does not seem to be used (last_log_time = follow_logs() is not used). Seems to be a remnant of previous version of log following. Can we just make the function -> None and do not return anything from it ?

Reply:
Scratch that - it is used. but still it seems rather strange as it is only used in calling the follow_logs(since_time=last_log_time) and it's Optional[datetime] there. This Union does not look right.

Reply:
As `parse_log_line` method return a pendulum object I thought to add before returning something like
```
if timestamp is not None:
    datetime.fromtimestamp(timestamp.timestamp())
```
Then we will have only the `Optional[datetime]` returning type.

Reply:
But I find another solution that reduces the number of types